src/librustc_trans/debuginfo/source_loc.rs

```diff
@@ -26,24 +27,24 @@ use syntax_pos::{Span, Pos};
 /// Sets the current debug location at the beginning of the span.
 ///
 /// Maps to a call to llvm::LLVMSetCurrentDebugLocation(...).
-pub fn set_source_location(fcx: &FunctionContext, builder: &Builder, scope: DIScope, span: Span) {
-    let function_debug_context = match fcx.debug_context {
+pub fn set_source_location(mir: &MirContext, builder: &Builder, scope: DIScope, span: Span) {
+    let function_debug_context = match mir.debug_context {
         FunctionDebugContext::DebugInfoDisabled => return,
         FunctionDebugContext::FunctionWithoutDebugInfo => {
-            set_debug_location(fcx.ccx, builder, UnknownLocation);
+            set_debug_location(mir.ccx(), builder, UnknownLocation);
             return;
         }
         FunctionDebugContext::RegularContext(ref data) => data
     };
 
     let dbg_loc = if function_debug_context.source_locations_enabled.get() {
-        debug!("set_source_location: {}", fcx.ccx.sess().codemap().span_to_string(span));
-        let loc = span_start(fcx.ccx, span);
+        debug!("set_source_location: {}", mir.ccx().sess().codemap().span_to_string(span));
+        let loc = span_start(mir.ccx(), span);
         InternalDebugLocation::new(scope, loc.line, loc.col.to_usize())
     } else {
         UnknownLocation
     };
-    set_debug_location(fcx.ccx, builder, dbg_loc);
+    set_debug_location(mir.ccx(), builder, dbg_loc);
```

Comment:
FWIW, here and a few times above, `builder` has a `ccx` of its own. `trans::debuginfo` should maybe take just `debug_context`, not the whole `mir` context.